apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/etl/FissTransformer.java

```diff
@@ -38,102 +39,168 @@
-          FissPayer.newBuilder().setBeneZPayer(payerBuilder.build()).build());
-
-      data.get(Fiss.ADMTG_DGNS_CD).ifPresent(claimBuilder::setAdmDiagCode);
-      consumeIf(
-          data.get(Fiss.CLM_FREQ_CD).orElse(null),
-          NumberUtils::isDigits,
-          value -> claimBuilder.setFreqCdEnumValue(Integer.parseInt(value)));
-      data.getFromType(Fiss.CLM_FRM_DT, Parser.Data.Type.DATE)
-          .ifPresent(claimBuilder::setStmtCovFromCymd);
-      consumeIf(
-          data.get(Fiss.CLM_SRVC_CLSFCTN_TYPE_CD).orElse(null),
-          NumberUtils::isDigits,
-          value -> claimBuilder.setServTypeCdEnumValue(Integer.parseInt(value)));
-      data.getFromType(Fiss.CLM_THRU_DT, Parser.Data.Type.DATE)
-          .ifPresent(claimBuilder::setStmtCovToCymd);
-      data.get(Fiss.CLM_TOT_CHRG_AMT).ifPresent(claimBuilder::setTotalChargeAmount);
-      consumeIf(
-          data.get(Fiss.CLM_FAC_TYPE_CD).orElse(null),
-          NumberUtils::isDigits,
-          value -> claimBuilder.setLobCdEnumValue(Integer.parseInt(value)));
-      data.get(Fiss.ORG_NPI_NUM).ifPresent(claimBuilder::setNpiNumber);
-      data.get(Fiss.PRNCPAL_DGNS_CD).ifPresent(claimBuilder::setPrincipleDiag);
-      data.get(Fiss.PRVDR_NUM)
-          .ifPresent(value -> claimBuilder.setMedaProv6(String.format("%.6s", value)));
-
-      addDiagCodes(claimBuilder, data);
-      addProcCodes(claimBuilder, data);
-
-      return FissClaimChange.newBuilder()
-          .setTimestamp(Timestamp.newBuilder().setSeconds(Instant.now().getEpochSecond()).build())
-          .setSeq(sequenceNumber.inc())
-          .setClaim(claimBuilder.build())
-          .setChangeType(ChangeType.CHANGE_TYPE_UPDATE)
-          .build();
+      if (message.getLineNumber() == (lineNumber - 1)) {
+        // If it's the next sequential line number, add to previous claim
+        message.setMessage(addToExistingClaim(storedClaim, data));
+        message.setLineNumber(lineNumber);
+        claimToReturn = null;
+      } else if (lineNumber == 1) {
+        // If the line number is 1, it's a new claim, return the old, store the new
+        claimToReturn = storedClaim;
+        message.setLineNumber(1);
+        message.setMessage(transformNewClaim(sequenceNumber, data, mbiSampler, sampleId));
+      } else {
+        // If it's not the next claim or a new one starting at 1, then something is wrong.
+        throw new IllegalStateException(
+            String.format(
+                "Invalid row sequence, previous line number: %d, current line number: %d",
+                message.getLineNumber(), lineNumber));
+      }
+    } else {
+      if (lineNumber != 1) {
+        throw new IllegalStateException(
+            String.format(
+                "Invalid row sequence, expected: 1, current line number: %d", lineNumber));
+      }
+
+      message.setLineNumber(lineNumber);
+      message.setMessage(transformNewClaim(sequenceNumber, data, mbiSampler, sampleId));
+      claimToReturn = null;
```

Comment:
If we are processing the last line of the file, does this need to be set to the stored claim?

Reply:
At the bridge level, after all the claims have been processed and the loop exits, it checks if there was a `WrappedMessage` that hasn't been processed yet.
https://github.com/CMSgov/beneficiary-fhir-data/blob/kylerobbins/paca-432-parse-mcs-line-items/apps/bfd-pipeline/bfd-pipeline-rda-bridge/src/main/java/gov/cms/bfd/pipeline/bridge/RDABridge.java#L303-L307